### ISSUE TITLE
Does not work properly

### ISSUE THREAD

**Issue by CrazyWolf13** [kitarolivier/KSoft.MediaInfo#2]
Would be a great plugin but sadly it doesn't catch up any media playing.
Other users from the discord said the same. @kitarolivier

**Comment by kitarolivier** [comment]
I am on holidays now. I'll check that later.

**Comment by CrazyWolf13** [comment]
> 
Thanks for the quick response, then i wish you nice holidays.

**Comment by kitarolivier** [comment]
@CrazyWolf13 : tell me more about your environement (Windows/Unix..) and which music player you use ? 

Did you checked that [thread](https://github.com/kitarolivier/KSoft.MediaInfo/issues/1) ?

**Comment by CrazyWolf13** [comment]
> @CrazyWolf13 : tell me more about your environement (Windows/Unix..) and which music player you use ?
> 
> Did you checked that [thread](https://github.com/kitarolivier/KSoft.MediaInfo/issues/1) ?

Yes i checked the thread.
I have got windows 10
I tested it with various music players in the browser and i can control them via media control in macropad and the keypad keys.
I tested twitch web, youtube music web youtube music app none worked.

**Comment by kitarolivier** [comment]
Sorry for the long delay for the answer.

I have updated macro deck and all plugins to latest version, and MediaInfo is working for Spotify (both web player and windows application), with Youtube (standard, not YT music). 

I also tested with Twitch web (first use ;) ), and it's working :
<img width="1314" alt="image" src="https://user-images.githubusercontent.com/37141868/196694446-c047504c-8357-4ad7-8f6b-ee0d8d49d239.png">

It doesn't work with Windows media player and VLC player.
Something strange : my control keys control playback and volume, but only volume is showing. No image, no music title or artist. Something has probably changed in windows ? I'm trying to inverstigate, but it can take some time.

**Comment by CrazyWolf13** [comment]
> Sorry for the long delay for the answer.
> 
> I have updated macro deck and all plugins to latest version, and MediaInfo is working for Spotify (both web player and windows application), with Youtube (standard, not YT music). 
> 
> I also tested with Twitch web (first use ;) ), and it's working :
> <img width="1314" alt="image" src="https://user-images.githubusercontent.com/37141868/196694446-c047504c-8357-4ad7-8f6b-ee0d8d49d239.png">
> 
> It doesn't work with Windows media player and VLC player.
> Something strange : my control keys control playback and volume, but only volume is showing. No image, no music title or artist. Something has probably changed in windows ? I'm trying to inverstigate, but it can take some time.
> 
> 
> 

I noticed its working better than before but still inconsistently.
Sometimes it does work sometimes not.

I also noticed that even with a if variable music is playing true set button to on its not always updating sometimes i have to press the button and only then it reloads.
And updates.

Its like sometimes it works sometimes not also if you have multiple audio sources playing even if you paused one its not updating to the new playing source i have to manually close the old application and restart the new one.

**Comment by kitarolivier** [comment]
Thanks for the Feedback.

If there are multiple sources, I take the first one. If you see that when using volume or controls
![image](https://user-images.githubusercontent.com/37141868/186089963-bf6bf17b-f2c0-4c14-ac35-ce9f307c7549.png)
when you click on the right arrow on the right in the gray rectangle, you can choose which audio sources you control. It seems. Its not a problem to add multiple variables with the sources (i.e. audio1=Youtube, audio2=Spotify, ...), but it could be more difficult to create a functional panel in macro deck to make use of these.
Multiple sources can be the cause of the inconstancy, because maybe you have a source with no sound, and it take this as the default, so nothing shows up.
Try to close sources so that there is not the right arrow on gray rectangle on the right, and tell me if it's more reliable.

Regards,
Olivier.

**Comment by CrazyWolf13** [comment]
> Thanks for the Feedback.
> 
> If there are multiple sources, I take the first one. If you see that when using volume or controls
> ![image](https://user-images.githubusercontent.com/37141868/186089963-bf6bf17b-f2c0-4c14-ac35-ce9f307c7549.png)
> when you click on the right arrow on the right in the gray rectangle, you can choose which audio sources you control. It seems. Its not a problem to add multiple variables with the sources (i.e. audio1=Youtube, audio2=Spotify, ...), but it could be more difficult to create a functional panel in macro deck to make use of these.
> Multiple sources can be the cause of the inconstancy, because maybe you have a source with no sound, and it take this as the default, so nothing shows up.
> Try to close sources so that there is not the right arrow on gray rectangle on the right, and tell me if it's more reliable.
> 
> Regards,
> Olivier.

Thanks for the explanation i will try to close them. But how could i resolve the issue with the button itself not updating until i click it?

**Comment by CrazyWolf13** [comment]
I just started my computer and Opera Youtube Music Web Player: 
![Screenshot 2022-10-20 124048](https://user-images.githubusercontent.com/96661824/196927386-ab77925b-46b5-4095-85f2-0834ec451bf4.jpg)
![Bild_2022-10-20_124119494](https://user-images.githubusercontent.com/96661824/196927394-73f337f0-ac31-4fb9-857d-e0a902bd4377.png)
![Screenshot 2022-10-20 124028](https://user-images.githubusercontent.com/96661824/196929826-c5c2472c-bbfa-472c-9fb0-97d81739269e.jpg)


Here with Twitch:
![Screenshot 2022-10-20 124405](https://user-images.githubusercontent.com/96661824/196928121-3d087223-3a04-4bf0-871d-1223faedf24c.jpg)
![Screenshot 2022-10-20 124355](https://user-images.githubusercontent.com/96661824/196928123-55aa1182-df2b-4dab-9db8-3b762e695555.jpg)
![Screenshot 2022-10-20 124342](https://user-images.githubusercontent.com/96661824/196928126-ae971c51-9fa7-462a-a14d-dd13e212c6ff.jpg)

Even with multiple Tabs playing music it does not work.

And a soon as I start another application and start playing music all the others appear : I start Youtube Music player on opera gx nothing shows up , i start spotify desktop: the youtube music web player song shows up perfectly fine.

And as soon as a new song starts it dissapears from the macro deck and i have to restart spotify and then the youtube music player songs gets displayed again. 

In conclusion i can say I need to open another music application for the macrodeck to display it. 

Hope those details help.
If you need more details no problem.
Regards,
Crazy_Wolf

**Comment by kitarolivier** [comment]
Thank you for the feedback.

I went to the same conclusion that Chrome (I use chrome) should be running to be able to get the information. I can't test Opera now, but chan you can test.

The plugin tries to make a connection to the service when it starts. If the connection fails, it won't try anymore. So starting Chrome or Opera should be done before launching macro desk. You can restart macro deck or disable and re-enable the plugin. Can you confirm ? I'll update the readme with the results.

I also will try to find some time to make a timer trying to reconnect if first connection fails (maybe in the settings, with an interval time, and retry count ?).

Regards,
Olivier.

**Comment by CrazyWolf13** [comment]
> Thank you for the feedback.
> 
> I went to the same conclusion that Chrome (I use chrome) should be running to be able to get the information. I can't test Opera now, but chan you can test.
> 
> The plugin tries to make a connection to the service when it starts. If the connection fails, it won't try anymore. So starting Chrome or Opera should be done before launching macro desk. You can restart macro deck or disable and re-enable the plugin. Can you confirm ? I'll update the readme with the results.
> 
> I also will try to find some time to make a timer trying to reconnect if first connection fails (maybe in the settings, with an interval time, and retry count ?).
> 
> Regards,
> Olivier.
> 

Its a bit confusing, as soon as a play music in opera nothing happens but as soon as i start spotify the plugins starts and displays the music from opera

**Comment by CrazyWolf13** [comment]
> Thank you for the feedback.
> 
> I went to the same conclusion that Chrome (I use chrome) should be running to be able to get the information. I can't test Opera now, but chan you can test.
> 
> The plugin tries to make a connection to the service when it starts. If the connection fails, it won't try anymore. So starting Chrome or Opera should be done before launching macro desk. You can restart macro deck or disable and re-enable the plugin. Can you confirm ? I'll update the readme with the results.
> 
> I also will try to find some time to make a timer trying to reconnect if first connection fails (maybe in the settings, with an interval time, and retry count ?).
> 
> Regards,
> Olivier.
> 

Maybe an easy fix would be zo add an action to reload the music fetcher, then i would configure this to always reload when the time variable changes.

**Comment by CrazyWolf13** [comment]
@kitarolivier  is this possible?

**Comment by kitarolivier** [comment]
Hello Crazy Wolf,

Sorry for the long delay, but I had almost no time the last month.

I will try to implement a timer and and "retry" button in the settings page before 2023 :)

**Comment by CrazyWolf13** [comment]
> Hello Crazy Wolf,
> 
> Sorry for the long delay, but I had almost no time the last month.
> 
> I will try to implement a timer and and "retry" button in the settings page before 2023 :)

Hey there, no problem take your time!

Maybe you could as well use the WebNowPlaying dll which servs exactly this purpose, its a browser addon and it's mainly used in Rainmeter. The project is as well on Github.

https://github.com/tjhrulz/WebNowPlaying

**Comment by kitarolivier** [comment]
I will check that :)
